src/storage/kura/mod.rs

```diff
@@ -0,0 +1,235 @@
+}
+
+use chashmap::CHashMap;
+/// WSV reflects the current state of the system, can be considered as a snapshot. For example, WSV
+/// holds information about an amount of assets that an account has at the moment but does not
+/// contain any info history of transaction flow.
+//TODO: can we assume that these queries https://iroha.readthedocs.io/en/latest/develop/api/queries.html
+//will be use cases to read from WSV?
+#[derive(Default)]
+struct WorldStateView {
+    /*Structure of arrays?*/
+    /// Map of `account_id` to vector of assets.
+    accounts_assets: CHashMap<String, Vec<model::Asset>>,
+    /// Map of `account_id` to vector of inbound transactions.
+    accounts_inbound_transactions: CHashMap<String, Vec<model::Transaction>>,
+    /// Map of `account_id` to vector of outbound transactions.
+    accounts_outbound_transactions: CHashMap<String, Vec<model::Transaction>>,
+    /// Map of `account_id` to vector of all transactions.
+    accounts_all_transactions: CHashMap<String, Vec<model::Transaction>>,
+    /// Map of `asset_id` to vector of all transactions.
+    assets_transactions: CHashMap<String, Vec<model::Transaction>>,
+}
+
+//TODO: rename
+enum Relation {
+    /// Belongs to account with defined identification.
+    /// For example we can fill a map of accounts to assets by this relation.
+    BelongsTo(String),
+    GoingTo(String),
+    GoingFrom(String),
+}
+
+/// This trait should be implemented for commands with `account_id` field.
+/// Marking your command with `impl` of this trait you provide an ability
+/// to retrieve information about relation to an account.
+//TODO: name is very bad, should be renamed.
+trait Accountability {
+    fn relations() -> Vec<Relation> {
+        unimplemented!();
+    }
+}
+
+impl WorldStateView {
+    fn put(&mut self, block: model::Block) {
+        self.accounts_assets = merge_accounts_assets(self.accounts_assets.clone(), block.clone());
+        self.accounts_inbound_transactions =
+            merge_inbound_transactions(self.accounts_inbound_transactions.clone(), block.clone());
+        self.accounts_outbound_transactions =
+            merge_outbound_transactions(self.accounts_outbound_transactions.clone(), block.clone());
+        self.accounts_all_transactions =
+            merge_all_transactions(self.accounts_all_transactions.clone(), block.clone());
+        self.assets_transactions =
+            merge_assets_transactions(self.assets_transactions.clone(), block.clone());
+    }
+
+    fn get_assets_by_account_id(&self, account_id: &str) -> Option<Vec<model::Asset>> {
+        match &self.accounts_assets.get(account_id) {
+            Some(assets) => Some(assets.clone().to_vec()),
+            None => None,
+        }
+    }
+}
+
+fn merge_accounts_assets(
+    origin: CHashMap<String, Vec<model::Asset>>,
+    block: model::Block,
+) -> CHashMap<String, Vec<model::Asset>> {
+    origin.insert(
+        block.transactions[0].account_id.clone(),
```

Comment:
I'm not sure I understand the logic here. Why transactions[0]? Also, won't this fail if there are 0 transactions?